tests_integ/helpers/base_test.py

```diff
@@ -0,0 +1,345 @@
+        return self.my_region
+
+    def get_s3_uri(self, file_name):
+        """
+        Returns the S3 URI of a resource file
+
+        Parameters
+        ----------
+        file_name : string
+            Resource file name
+        """
+        return FILE_TO_S3_URL_MAP[file_name]["url"]
+
+    def get_code_key_s3_uri(self, code_key):
+        """
+        Returns the S3 URI of a code key for template replacement
+
+        Parameters
+        ----------
+        code_key : string
+            Template code key
+        """
+        return FILE_TO_S3_URL_MAP[CODE_KEY_TO_FILE_MAP[code_key]]["url"]
+
+    def get_stack_resources(self, resource_type, stack_resources=None):
+        if not stack_resources:
+            stack_resources = self.stack_resources
+
+        resources = []
+        for res in stack_resources["StackResourceSummaries"]:
+            if res["ResourceType"] == resource_type:
+                resources.append(res)
+
+        return resources
+
+    def get_stack_output(self, output_key):
+        for output in self.stack_description["Stacks"][0]["Outputs"]:
+            if output["OutputKey"] == output_key:
+                return output
+        return None
+
+    def get_stack_tags(self, output_name):
+        resource_arn = self.get_stack_output(output_name)["OutputValue"]
+        return self.sfn_client.list_tags_for_resource(resourceArn=resource_arn)["tags"]
+
+    def get_stack_deployment_ids(self):
+        resources = self.get_stack_resources("AWS::ApiGateway::Deployment")
+        ids = []
+        for res in resources:
+            ids.append(res["LogicalResourceId"])
+
+        return ids
+
+    def get_stack_stages(self):
+        resources = self.get_stack_resources("AWS::ApiGateway::RestApi")
+
+        if not resources:
+            return []
+
+        return self.api_client.get_stages(restApiId=resources[0]["PhysicalResourceId"])["item"]
+
+    def get_stack_stages_v2(self):
+        resources = self.get_stack_resources("AWS::ApiGatewayV2::Api")
+
+        if not resources:
+            return None
+
+        return self.api_v2_client.get_stages(ApiId=resources[0]["PhysicalResourceId"])["Items"]
+
+    def get_stack_nested_stack_resources(self):
+        resources = self.get_stack_resources("AWS::CloudFormation::Stack")
+
+        if not resources:
+            return None
+
+        return self.cloudformation_client.list_stack_resources(StackName=resources[0]["PhysicalResourceId"])
+
+    def get_stack_outputs(self):
+        if not self.stack_description:
+            return {}
+        else:
+            output_key_to_value = {}
+            output_list = self.stack_description["Stacks"][0]["Outputs"]
+            for output in output_list:
+                output_key_to_value[output["OutputKey"]] = output["OutputValue"]
+            return output_key_to_value
+
+    def get_resource_status_by_logical_id(self, logical_id):
+        if not self.stack_resources:
+            return ""
```

Comment:
Any reason for returning an empty string instead of `None`?

Reply:
Changing to None